akka-docs/src/main/paradox/stream/operators/Sink/foreachAsync.md

```diff
@@ -0,0 +1,33 @@
+# foreachAsync
+
+Invoke a given procedure asynchronously for each element received.
+
+@ref[Sink stages](../index.md#sink-stages)
+
+@@@div { .group-scala }
+
+## Signature
+
+@@signature [Sink.scala]($akka$/akka-stream/src/main/scala/akka/stream/scaladsl/Sink.scala) { #foreachAsync }
+
+@@@
+
+## Description
+
+Invoke a given procedure asynchronously for each element received. Note that it is not safe to mutate shared state from the procedure.
```

Comment:
I'm not sure about the wording here. The purpose of `foreach` is to do side effects and that typically involves mutating state somewhere. Perhaps: "Note that if shared state is mutated from the procedure that must be done in a thread-safe way."